package/kernel/linux/modules/netdevices.mk

```diff
@@ -1042,3 +1042,53 @@ define KernelPackage/be2net/description
 endef
 
 $(eval $(call KernelPackage,be2net))
+
+
+define KernelPackage/qca7k
+  SUBMENU:=$(NETWORK_DEVICES_MENU)
+  TITLE:=Qualcomm Atheros QCA7000 support (library for SPI/UART driver)
+  HIDDEN:=1
+  FILES:=$(LINUX_DIR)/drivers/net/ethernet/qualcomm/qca_7k_common.ko
+  KCONFIG:=CONFIG_QCA7000
+endef
+
+define KernelPackage/qca7k/description
+  Kernel module for shared code for Qualcomm Atheros QCA7000 SPI/UART drivers
+endef
+
+$(eval $(call KernelPackage,qca7k))
+
+
+define KernelPackage/qcaspi
+  SUBMENU:=$(NETWORK_DEVICES_MENU)
+  TITLE:=Qualcomm Atheros QCA7000 SPI support
+  DEPENDS:=+kmod-qca7k
+  FILES:=$(LINUX_DIR)/drivers/net/ethernet/qualcomm/qcaspi.ko
+  KCONFIG:= \
+	CONFIG_QCA7000_SPI \
+	CONFIG_SPI=y \
+	CONFIG_SPI_MASTER=y
```

Comment:
Targets that support `SPI` will already have `SPI` set to y, so no need to force this to y for targets where it won't be able to run anyway.

Reply:
I copied from spi-ks8995 module, then this should be removed there, too?
You mean that platform which support SPI already set `CONFIG_SPI=y` and `CONFIG_SPI_MASTER=y` in `target/linux/xyz/config-*` ?
Then I could drop `SPI_MASTER` also, right?

Reply:
Yes, as `SPI` implies `SPI_MASTER`.

What we basically are missing is a way to say "target has SPI support", so the drivers can depend on that. Like we do for PCI and USB support.